src/holidata/holidays/SG.py

```diff
@@ -0,0 +1,223 @@
+from datetime import date, timedelta
```

Comment:
As mentioned in the comment to `shift_to_monday_if_weekend`, use `SmartDayArrow` for all dates, i.e. do not use the datetime module at all. Also here, sorry for my imprecision when [I just said](https://github.com/GothenburgBitFactory/holidata/pull/108#discussion_r1703394149) "_Use holidatas class SmartDayArrow..._"

See other country files on how to this is done, e.g. the [region file for Ceuta](https://github.com/GothenburgBitFactory/holidata/blob/4bff11a0249fac9851637a4128e0a2639a79d020/src/holidata/holidays/ES/CE.py#L82) where a date map similar to those used by you here is defined, or the [country file for Great Britain](https://github.com/GothenburgBitFactory/holidata/blob/main/src/holidata/holidays/GB.py).